bridge/bridge/deploy/local.py

```diff
@@ -0,0 +1,283 @@
+from typing import List, Dict, Set
+from bridge.types import Model, ModelVersion, Artifact
+from bridge.deploy import DeployTarget
+import socket
+import signal
+import os
+from subprocess import Popen
+from collections import defaultdict
+import logging
+from pprint import pformat
+import threading
+from flask import Flask, request, Response
+import requests
+
+logger = logging.getLogger(__name__)
+CONDA_LOCK_FILE = ".brdg_local.lock"
+
+
+class LocalDeploymentProxy:
+    def __init__(self, target):
+        self.target = target
+        self.app = Flask(__name__)
+        self.app.add_url_rule(
+            "/<model>/<stage>/<path:path>",
+            view_func=self.proxy,
+            methods=["POST"],
+        )
+
+    def proxy(self, model, stage, path):
+        port = self.target.routes.get(model, {}).get(stage)
+
+        if port is None:
+            return 404
+
+        resp = requests.request(
+            method=request.method,
+            url=f"http://localhost:{port}/{path}",
+            headers={
+                key: value for (key, value) in request.headers if key != "Host"
+            },
+            data=request.get_data(),
+            cookies=request.cookies,
+            allow_redirects=False,
+            params=dict(request.args),
+        )
+
+        excluded_headers = [
+            "content-encoding",
+            "content-length",
+            "transfer-encoding",
+            "connection",
+        ]
+        headers = [
+            (name, value)
+            for (name, value) in resp.raw.headers.items()
+            if name.lower() not in excluded_headers
+        ]
+
+        response = Response(resp.content, resp.status_code, headers)
+        return response
+
+    def run(self):
+        port = os.environ.get("PORT", 3000)
+        self.handle = threading.Thread(
+            target=self.app.run, kwargs={"host": "0.0.0.0", "port": port}
+        )
+        self.handle.start()
+
+
+def destroy_failed_conda(model_dir):
```

Comment:
nit: could cal this just destroy_conda and then also use it to prune cached environments when we eventually need to do that under disk pressure

Reply:
I can rename it if we implement that functionality, for now it is only used if the install process or the model exits with an error.